Casks/o/okta-verify.rb

```diff
@@ -0,0 +1,47 @@
+cask "okta-verify" do
+  version "9.52.0,5785-cfe7e6a"
+  sha256 "b03d47fbdda0be998a54034471f1dc071ef65deea909784d4c1a37f8cf57b23f"
+
+  url "https://okta.okta.com/artifacts/OKTA_VERIFY_MACOS/#{version.csv.first}/OktaVerify-#{version.csv.first}-#{version.csv.second}.pkg"
+  name "Okta Verify"
+  desc "Identity verification provider"
+  homepage "https://help.okta.com/eu/en-us/content/topics/end-user/ov-overview-macos.htm"
+
+  livecheck do
+    url "https://okta.okta.com/api/v1/artifacts/OKTA_VERIFY_MACOS/latest?releasesChannel=GA"
+    strategy :json do |json|
+      version_part = json["version"]
+      href = json.dig("files", 0, "href")
+      build_part = href.match(/OktaVerify-.*?-(.*?).pkg/i)&.captures&.first
+      "#{version_part},#{build_part}"
```

Comment:
```suggestion
    regex(/OktaVerify[._-]v?(\d+(?:\.\d+)+)(?:[._-](.+))?\.pkg/i)
    strategy :json do |json, regex|
      json["files"]&.filter_map do |file|
        match = file["href"]&.match(regex)
        next unless match

        match[2] ? "#{match[1]},#{match[2]}" : match[1]
      end
```

* The version part can be parsed from the file name, so it's simpler to handle that with the regex.
* The regex should be set in a `regex()` call and passed into the `strategy block`.
* We should not assume that the first member in the `files` array will be the file we want. Instead, we should iterate through the files array and work with any matching URLs.
* I've set this up to make the suffix after the version to be optional, so livecheck will (hopefully) still surface a new version and we can manually adjust the cask.